### ISSUE TITLE
Does usage data persist on store limiters, when delete rate limiter instance and create new one with the same `keyPrefix`?

### ISSUE THREAD

**Issue by Jaymoss520** [animir/node-rate-limiter-flexible#45]
If I create a new instance of the rate limiter, will it remember past usage info (specifically, if I pass `keyPrefix`), or will it begin fresh and think there has been no usage so far? E.g. say I have an instance that consumes 50 points, then it gets deleted. If I create a new instance before the duration expires, will it remember that 50 points were recently consumed?

Also, does each instance create a separate connection to redis?

Basically, my use-case requires different rate-limiting configurations for each client, so if I were to use this service, I would need a different rate limiter instance for each client. This is okay if I can create/delete the instances if the data persists in redis. But if the instances cannot be deleted, I worry if each instance will have their own redis connection (which can cause issues).

**Comment by animir** [comment]
@Jaymoss520 
If you create an instance with the same `keyPrefix` and consume points for the same `key` before duration expires, yes, it remembers consumed points. Think of limiter as interface to access counter value by key in store.

Every time rate limiter instance is created, `storeClient` should be provided for Redis. It may be the same Redis client for all your limiters. It is up to you.

Could you provide two examples of rate limiter options for different clients?

**Comment by animir** [comment]
@Jaymoss520 I'm closing this. Feel free to re-open, if you still have questions.
